src/util/std_expr.h

```diff
@@ -1872,11 +1872,6 @@ inline union_exprt &to_union_expr(exprt &expr)
 class struct_exprt : public multi_ary_exprt
 {
 public:
-  DEPRECATED("use struct_exprt(component_name, value, type) instead")
-  struct_exprt() : multi_ary_exprt(ID_struct)
-  {
-  }
```

Comment:
Do we apply a 6-months rule?

Reply:
I would say it really depends on how much it inconveniences others. This one is a bit risky, since structs are pretty common. On the other hand, it's likely easy to fix.

Reply:
Well, _something_ in this PR does seem to break building TG, so clearly one of the constructors is being used externally. (#3768 does take care of removing all of them, but is marked as do-not-merge-before-end-of-March.)

Reply:
That might well be a reasonable indicator. Let's wait with this one, and I'll merge #4367.